packages/wdio-cli/src/run.js

```diff
@@ -64,9 +65,24 @@ export default function run (params) {
     })
 }
 
-function launch (wdioConf, params) {
+/**
+ * launcher callback function
+ * @param {number|undefined}    exitCode    exit code
+ * @param {Error|undefined}     err         Error
+ * @param {boolean|undefined}   isComplete  if all tests and hooks were executed
+ */
+export const launchCallback = (exitCode, err, isComplete) => {
+    if (!err) {
+        return { exitCode }
+    }
+    console.error(err)
+    if (!isComplete) {
+        process.emit('shutdown', 1)
+    }
+    return { err, isComplete }
+}
+
+export function launch (wdioConf, params) {
     const launcher = new Launcher(wdioConf, params)
-    launcher.run().then(
-        (code) => process.nextTick(() => process.exit(code)),
-        (e) => process.nextTick(() => { throw e }))
+    launcher.run(launchCallback)
```

Comment:
`run` is async function, we need to resolve promise

Reply:
We use callback instead

Reply:
I can change implementation to use promises instead of callback

Reply:
I think it would be more convenient.

Reply:
Hey, sorry for confusing, we are not waiting for any result from async function, so there is no need in then/catch here